datadog/resource_datadog_synthetics_test_.go

```diff
@@ -0,0 +1,432 @@
+			locations = append(locations, s.(string))
+		}
+		syntheticsTest.Locations = locations
+	}
+
+	if attr, ok := d.GetOk("tags"); ok {
+		tags := []string{}
+		for _, s := range attr.([]interface{}) {
+			tags = append(tags, s.(string))
+		}
+		syntheticsTest.Tags = tags
+	}
+
+	return &syntheticsTest
+}
+
+func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *datadog.SyntheticsTest) {
+	d.Set("type", syntheticsTest.GetType())
+
+	actualRequest := syntheticsTest.GetConfig().Request
+	localRequest := newLocalMap(map[string]interface{}{
+		"method":  actualRequest.GetMethod(),
+		"url":     actualRequest.GetUrl(),
+		"body":    actualRequest.GetBody(),
+		"timeout": actualRequest.GetTimeout(),
+	})
+	d.Set("request", localRequest)
+	d.Set("request_headers", actualRequest.Headers)
+
+	actualAssertions := syntheticsTest.GetConfig().Assertions
+	localAssertions := []map[string]string{}
+	for _, assertion := range actualAssertions {
+		localAssertion := newLocalMap(map[string]interface{}{
+			"type":     assertion.GetType(),
+			"property": assertion.GetProperty(),
+			"operator": assertion.GetOperator(),
+			"target":   assertion.Target,
+		})
+		localAssertions = append(localAssertions, localAssertion)
+	}
+	d.Set("assertions", localAssertions)
+
+	actualDevices := syntheticsTest.GetOptions().Devices
+	localDevices := []map[string]string{}
+	for _, device := range actualDevices {
+		localDevice := newLocalMap(map[string]interface{}{
+			"id":       device.GetId(),
+			"name":     device.GetName(),
+			"height":   device.GetHeight(),
+			"width":    device.GetWidth(),
+			"isMobile": device.GetIsMobile(),
+		})
+		localDevices = append(localDevices, localDevice)
+	}
+	d.Set("devices", localDevices)
+
+	d.Set("locations", syntheticsTest.Locations)
+
+	actualOptions := syntheticsTest.GetOptions()
+	localOptions := newLocalMap(map[string]interface{}{
+		"tick_every":           actualOptions.GetTickEvery(),
+		"min_failure_duration": actualOptions.GetMinFailureDuration(),
+		"min_location_failed":  actualOptions.GetMinLocationFailed(),
+	})
+	d.Set("options", localOptions)
+
+	d.Set("name", syntheticsTest.GetName())
+	d.Set("message", syntheticsTest.GetMessage())
+	d.Set("tags", syntheticsTest.Tags)
+	d.Set("paused", *syntheticsTest.Status == "paused")
+}
+
+func updateSyntheticsTestLiveness(d *schema.ResourceData, client *datadog.Client) {
+	paused, ok := d.GetOk("paused")
+	if !ok {
+		return
+	}
+	if paused.(bool) {
+		client.PauseSyntheticsTest(d.Id())
+	} else {
+		client.ResumeSyntheticsTest(d.Id())
+	}
+}
+
+func newLocalMap(actualMap map[string]interface{}) map[string]string {
+	localMap := make(map[string]string)
+	for k, i := range actualMap {
+		var valStr string
+		switch v := i.(type) {
+		case bool:
+			if v {
+				valStr = "1"
+			} else {
+				valStr = "0"
+			}
+		case int:
+			valStr = strconv.Itoa(v)
+		case float64:
+			valStr = strconv.Itoa(int(v))
+		case string:
+			valStr = v
+		default:
+			// Ignore value
```

Comment:
I think it'd be better to do something like `valStr = string(v)`. Ignoring the default case could lead to some weird errors.

Reply:
Changed to:
```golang
// TODO: manage target for JSON body assertions
valStrr, err := json.Marshal(v)
if err == nil {
    valStr = string(valStrr)
}
```